File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/transtale/TranslateDb.java
package tw.nekomimi.nekogram.transtale;
import java.util.Locale;
import tw.nekomimi.nekogram.cc.CCTarget;
public class TranslateDb {
    public static TranslateDb currentTarget() { return new TranslateDb(); }
    public static TranslateDb currentInputTarget() { return new TranslateDb(); }
    public static TranslateDb forLocale(Locale locale) { return new TranslateDb(); }
    public static Locale getChatLanguage(long chatId, Locale defaultLocale) { return defaultLocale; }
    public static String getChatCCTarget(long chatId, String lang) { return null; }
    public static void clearAll() {}
    public boolean contains(String key) { return false; }
    public String query(String key) { return null; }
    public TranslateDb get(CCTarget target) { return this; }
    public String convert(String text) { return text; }
    public void save(String key, String val) {}
    public void clear() {}
}

File: TMessagesProj/src/main/java/tw/nekomimi/nekogram/transtale/TranslateDb.kt
package tw.nekomimi.nekogram.transtale

import java.util.*

class TranslateDb(val code: String) {
    companion object {
        @JvmStatic
        fun getChatLanguage(chatId: Long, default: Locale): Locale = default

        @JvmStatic
        fun saveChatLanguage(chatId: Long, locale: Locale) { /* no-op */ }

        @JvmStatic
        fun getChatCCTarget(chatId: Long, default: String?): String? = default

        @JvmStatic
        fun saveChatCCTarget(chatId: Long, target: String) { /* no-op */ }

        @JvmStatic
        fun currentTarget(): TranslateDb = TranslateDb("en")

        @JvmStatic
        fun forLocale(locale: Locale): TranslateDb = TranslateDb(locale.language)

        @JvmStatic
        fun currentInputTarget(): TranslateDb = TranslateDb("en")

        @JvmStatic
        fun clearAll() { /* no-op */ }
    }

    fun clear() { /* no-op */ }
    fun contains(text: String): Boolean = false
    fun query(text: String): String? = null
    fun save(text: String, trans: String) { /* no-op */ }
}

// Extension kept for compatibility
val Locale.transDb: TranslateDb get() = TranslateDb.forLocale(this)
val String.transDbByCode: TranslateDb get() = TranslateDb.forLocale(code2Locale)
